daemon/log.c

```diff
@@ -73,12 +71,7 @@ static FILE                 *fpLogfile = NULL;
 static MTC_BOOLEAN          initialized = FALSE;
 
 
-#if USE_SEMAPHORE
-#define MTC_LOG_SEMAPHORE_NAME  "/xha_log_semaphore"
-sem_t                       *semaphore = SEM_FAILED;
-#else
-static pthread_spinlock_t   lock;
-#endif
+static pthread_rwlock_t     lock;
```

Comment:
I think it is always helpful to have a comment describing the purpose of a lock and how it should be used.

In this case, the lock is to protect fpLogfile. Readers should hold the lock when accessing fpLogfile and the FILE it points to. Writers should hold the lock when assigning fpLogfile or when freeing the object it points to.